include/deal.II/lac/petsc_ts.h

```diff
@@ -0,0 +1,706 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2022 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE.md at
+//    the top level directory of deal.II.
+//
+//---------------------------------------------------------------
+//
+// Author: Stefano Zampini, King Abdullah University of Science and Technology.
+
+#ifndef dealii_petsc_ts_h
+#define dealii_petsc_ts_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_PETSC
+#  include <deal.II/base/mpi.h>
+#  include <deal.II/base/parameter_handler.h>
+#  include <deal.II/base/smartpointer.h>
+
+#  include <deal.II/lac/petsc_matrix_base.h>
+#  include <deal.II/lac/petsc_precondition.h>
+#  include <deal.II/lac/petsc_vector_base.h>
+
+#  include <petscdm.h>
+#  include <petscts.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+// Shorthand notation for PETSc error codes.
+#  define AssertTS(code)                                 \
+    do                                                   \
+      {                                                  \
+        PetscErrorCode __ierr = (code);                  \
+        AssertThrow(__ierr == 0, ExcPETScError(__ierr)); \
+      }                                                  \
+    while (0)
+
+namespace PETScWrappers
+{
+  /**
+   * Additional parameters that can be passed to the TimeStepper class.
+   */
+  class TimeStepperData
+  {
+  public:
+    /**
+     * Initialization parameters for TimeStepper.
+     *
+     * Running parameters:
+     *
+     * @param tstype Solver type
+     * @param initial_time Initial time
+     * @param final_time Final time
+     * @param initial_step_size Initial step size
+     * @param max_steps maximum number of steps
+     * @param match_step match requested final time
+     *
+     * Error parameters:
+     *
+     * @param minimum_step_size Minimum step size
+     * @param maximum_step_size Maximum step size
+     * @param absolute_tolerance Absolute error tolerance
+     * @param relative_tolerance Relative error tolerance
```

Comment:
You have some special semantics about these flags, see at least two of my comments below. These semantics should be documented here.